doc/syntax.md

```diff
@@ -0,0 +1,476 @@
+* `equations` - This block contains shape definitions and differential equations. It will be explained in further detail [later on in the manual](#equations).
+* `input` - This block is composed of one or more input ports. It will be explained in further detail [later on in the manual](#input).
+* `output` *`<event_type>`* - Defines which type of event the neuron can send. Currently, only `spike` is supported. No `end` is necessary at the end of this block.
+* `update` - Inside this block arbitrary code can be implemented using the internal programming language. The `update` block defines the runtime behaviour of the neuron. It contains the logic for state and equation [updates](#equations) and [refractoriness](#concepts-for-refractoriness). This block is translated into the `update` method in NEST.
+
+## Neuronal interactions
+
+### Synaptic input
+
+A neuron model written in NESTML can be configured to receive two distinct types of input, spikes and currents. For either of them, the modeler has to decide if inhibitory and excitatory inputs are lumped together into a single named buffer, or if they should be separated into different named buffers based on their sign. The `input` block is composed of one or more lines to express the exact combinations desired. Each line has the following general form:
+```
+port_name <- inhibitory? excitatory? (spike | current)
+```
+
+This way, a flexible combination of the inputs is possible. If, for example, current input should be lumped together, but spike input should be separated for inhibitory and excitatory incoming spikes, the following `input` block would be appropriate:
+```
+input:
+  currents <- current
+  inh_spikes <- inhibitory spike
+  exc_spikes <- excitatory spike
+end
+```
+
+Please note that it is equivalent if either both `inhibitory` and `excitatory` are given or none of them at all. If only a single one of them is given, another line has to be present and specify the inverse keyword. Failure to do so will result in a translation error.
+
+If there is more than one line specifying a `spike` or `current` port with the same sign, a neuron with multiple receptor types is created. In this case, a `receptor_types` entry is created in the status dictionary, which maps port names to numeric port indices in NEST. The receptor type can then be selected in NEST during [connection setup](http://nest-simulator.org/connection_management/#receptor-types).
+
+### Output
+
+Each neuron model can only send a single type of event. The type of the event has to be given in the `output` block. Currently, however, only spike output is supported.
+```
+output: spike
+```
+
+Please note that this block is *not* terminated with the `end` keyword.
+
+## Synaptic input
+
```

Comment:
@ingablundell: I think that we really should have some examples in this section to demonstrate the use of the functions. We should also have links to the models in the repository that use these concepts.